wave_vision/include/wave/vision/descriptor/orb_descriptor.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file
+ * ORB Descriptor Extractor implementation, derived from DescriptorExtractor
+ * base class.
+ * @ingroup vision
+ */
+#ifndef WAVE_VISION_ORB_DESCRIPTOR_HPP
+#define WAVE_VISION_ORB_DESCRIPTOR_HPP
+
+#include <string>
+#include <vector>
+
+#include "wave/vision/descriptor/descriptor_extractor.hpp"
+
+namespace wave {
+/** @addtogroup vision
+ *  @{ */
+
+/** Configuration parameters for the ORB descriptor extractor.
+ *
+ *  The full description of each parameter can be found
+ *  [here][ORBDescriptorParams]. Note: the ORBDescriptorParams struct only
+ *  consists of the parameters required for feature description, not detection.
+ *  ORB Feature Detector parameters can be found under ORBDetectorParams.
+ *
+ *  [ORBDescriptorParams]:
+ *  http://docs.opencv.org/trunk/db/d95/classcv_1_1ORB.html#adc371099dc902a9674bd98936e79739c
+ */
+struct ORBDescriptorParams {
+    ORBDescriptorParams() {}
+
+    ORBDescriptorParams(int wta_k, int patch_size)
+        : wta_k(wta_k), patch_size(patch_size), edge_threshold(patch_size) {}
+
+    /** Constructor using parameters extracted from a configuration file.
+     *
+     *  @param config_path the path to the location of the configuration file.
+     */
+    ORBDescriptorParams(const std::string &config_path);
+
+    /** The number of points that compose the ORB descriptor. A value of 2
+     *  constructs the descriptor by comparing the brightnesses of a random pair
+     *  of points. Other values are 3 or 4; 3 will use 3 random points, and 4
+     *  similarly will use 4 random points. For these last two options, the
+     *  output result will be 2 bits, as the result is the index of the
+     *  brightest point. Therefore, the distance method used for the
+     *  BruteForceMatcher or FLANNMatcher will have to be cv::NORM_HAMMING_2, a
+     *  variation to the Hamming distance, when WTA_K = 3 or 4.
+     *
+     *  Options: 2, 3, or 4.
+     *
+     *  Default: 2.
+     */
+    int wta_k = 2;
+
+    /** The size of the square patch used in the random point sampling to
+     *  construct the descriptor. This patch is smoothed using an integral
+     *  image.
+     *
+     *  Default: 31 (31 x 31 pixels). Must be greater than 0.
+     */
+    int patch_size = 31;
+
+    // Default parameters that should not be modified
```

Comment:
Yeah, exactly that @leokoppel . The thing is that they're required when I call the `cv::ORB::create()` method in the `ORBDescriptor` constructor. The detector has the opposite...only `wta_k` and `patch_size` are defined as "do not touch".

I don't think I want to keep these values defined in the `ORBDescriptor` constructor...should I just make these `Params` structs into classes and protect them? They have constructor methods, so making them classes wouldn't be the silliest idea.